google/cloud/pubsublite/internal/service.h

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUBLITE_INTERNAL_SERVICE_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUBLITE_INTERNAL_SERVICE_H
+
+#include "google/cloud/future.h"
+#include "google/cloud/status.h"
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace pubsublite_internal {
+
+/**
+ * A `Service` object is one that has a `Start` method that must be called
```

Comment:
Please describe what the class represents, not the functions it has, I can read those.  Comments are for humans, they should explain the intent.  Maybe something like this?

```cc
/**
  * A `Service` is anything that can start, run for a while, possibly with an error, and then can shutdown.
  *
  * Examples include resumable streaming RPCs of different types, and compositions of
  * these streaming RPCs.
  */
```